slither/core/dominators/utils.py

```diff
@@ -9,9 +9,16 @@
 def intersection_predecessor(node: "Node") -> Set["Node"]:
     if not node.fathers:
         return set()
-    ret = node.fathers[0].dominators
-    for pred in node.fathers[1:]:
-        ret = ret.intersection(pred.dominators)
+    if not any(father.is_reachable for father in node.fathers):
+        return set()
+
+    ret = set()
+    for pred in node.fathers:
+        ret = ret.union(pred.dominators)
```

Comment:
Mhh I am confused, can you explain why are we doing the union over the predecessor dominators, and then their interaction?

It seem to me that this loop should be removed - unless I am missing something here

Reply:
It's the safest way to start set intersections. I could store the first set as the base and start interacting all sets with that but the first set may not exist. So the safest way is to get every element from every set, and start to intersect every set with that big set

Reply:
gotcha that makes sense :)